MantidQt/MantidWidgets/inc/MantidQtMantidWidgets/InputController.h

```diff
@@ -245,29 +245,59 @@ class EXPORT_OPT_MANTIDQT_MANTIDWIDGETS InputControllerDraw
 /**
     Controller for erasing peaks on an unwrapped surface.
   */
-class EXPORT_OPT_MANTIDQT_MANTIDWIDGETS InputControllerErase
+class EXPORT_OPT_MANTIDQT_MANTIDWIDGETS InputControllerSelection
     : public InputControllerDraw {
   Q_OBJECT
 
 public:
-  InputControllerErase(QObject *parent);
-  ~InputControllerErase() override;
+  InputControllerSelection(QObject *parent);
+  ~InputControllerSelection() override;
   void onPaint(QPainter &) override;
 
-signals:
-  void erase(const QRect &);
-
 private:
   void drawCursor(QPixmap *cursor) override;
-  void signalLeftClick() override;
   void setPosition(const QPoint &pos) override;
   void resize() override;
 
-  QRect m_rect;
+protected:
```

Comment:
https://github.com/mantidproject/mantid/blob/51daa1c0424fd6d8dc79f1ed7b350943e70f1123/MantidQt/MantidWidgets/inc/MantidQtMantidWidgets/InputController.h#L328

Could we have the base class have a virtual method that the derived classes implement here. The method could take the `QRect` as a parameter and this way it could be `private` in the base class?